### PR TITLE
Added GitHub Actions for testing

### LINKED ISSUE #19: Switch from unworked Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by grogy**
It replaced unused TravisCI. I inspired in [PHP-Parallel-Lin](https://github.com/php-parallel-lint/PHP-Parallel-Lint/blob/c510dfa8a531229d85c730a19c20b1da08ee35f4/.github/workflows/test.yml) from @jrfnl 

Fixes #19

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - develop
+      - master
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Run tests on PHP ${{ matrix.php }}
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental == true }}
```
Comment (jrfnl) [inline .github/workflows/test.yml]: There is no `experimental` key in use in the test matrix, so this line is redundant as-is.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      matrix:
+        php:
+          - '5.4'
+          - '5.5'
+          - '5.6'
+          - '7.0'
+          - '7.1'
+          - '7.2'
+          - '7.3'
+          - '7.4'
```
Comment (jrfnl) [inline .github/workflows/test.yml:30]: What about adding PHP `8.0`, `8.1` and `8.2` ?

For PHP 8.2, you could then change the `continue-on-error: ${{ matrix.experimental == true }}` setting to: `continue-on-error: ${{ matrix.php == '8.2' }}` to allow builds against PHP 8.2 to fail while it is still under development.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          ini-values: error_reporting=E_ALL, display_errors=On
```
Comment (jrfnl) [inline .github/workflows/test.yml]:
```suggestion
          ini-values: zend.assertions=1, error_reporting=-1, display_errors=On
```

* `E_ALL` does not contain **_all_** errors on all PHP versions, `-1` does.
* Assertions _may_ be in use in the tests, so never a bad thing to enable them when the CI includes a test run.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          coverage: none
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v1
+
+      - name: 'Linting code'
+        run: ./vendor/bin/parallel-lint --exclude vendor .
```
Comment (jrfnl) [inline .github/workflows/test.yml]:
```suggestion
        run: ./vendor/bin/parallel-lint --exclude vendor --exclude git .
```

(generally makes the run slightly faster as the `git` directory can be large)

Comment (jrfnl) [reply .github/workflows/test.yml]: You may also want to consider adding the `cs2pr` tool in the `setup-php` step and then adding `--checkstyle | cs2pr` to this command to allow results to be displayed inline in PRs.

**Review by jrfnl — COMMENTED** [review]
@grogy Nicely done.

I've left a couple of suggestions inline.

I'd also like to suggest adding the following to cancel older builds for the same branch when a new commit has been pushed:

```yaml
# Cancels all previous workflow runs for the same branch that have not yet completed.
concurrency:
  # The concurrency group contains the workflow name and the branch name.
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          ini-values: error_reporting=E_ALL, display_errors=On
+          coverage: none
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v1
```
Comment (jrfnl) [inline .github/workflows/test.yml]:
```suggestion
        uses: ramsey/composer-install@v2
```

(v2 was released in December)

File: .github/workflows/test.yml (hunk 2 above)
Comment (grogy) [reply .github/workflows/test.yml:30]: PHP 8.x will be added in next step. I want convert 1:1 state previous TravisCI state

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          ini-values: zend.assertions=1, error_reporting=-1, display_errors=On
+          coverage: none
```
Comment (jrfnl) [inline .github/workflows/test.yml:41]:
```suggestion
          coverage: none
          tools: cs2pr
```

(cs2pr won't work unless installed ;-))

File: .github/workflows/test.yml (hunk 2 above)
Comment (grogy) [reply .github/workflows/test.yml:30]: For 8.x is necessary add support, via https://github.com/php-parallel-lint/PHP-Var-Dump-Check/runs/5065543774?check_suite_focus=true

**Comment by grogy** [comment]
Please, can you review fixes?

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
+name: Run unit and style tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - develop
+      - master
```
Comment (jrfnl) [inline .github/workflows/test.yml]: Why are you using branch selection here ? That wasn't present in the Travis config ?

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,57 @@
+name: Run unit and style tests
```
Comment (jrfnl) [inline .github/workflows/test.yml]: I don't see any style checks in the script ? Is this the right name for the script ?

File: .github/workflows/test.yml (hunk 2 above)
Comment (jrfnl) [reply .github/workflows/test.yml:30]: That's fair enough. I'm actually tempted to suggest abandoning this repo in favour of a PHP_CodeSniffer sniff for the same at some point in the future. Should make PHP cross-version support easier ;-)

**Review by jrfnl — COMMENTED** [review]
Two questions left, but other than that, A-okay!

File: .github/workflows/test.yml (hunk 7 above)
Comment (grogy) [reply .github/workflows/test.yml]: The reason is final check after merge into master. Github Action describes it in https://github.community/t/trigger-workflow-only-on-pull-request-merge/17359/3

File: .github/workflows/test.yml (hunk 8 above)
Comment (grogy) [reply .github/workflows/test.yml]: Is not correct. Fixed :+1:

File: .github/workflows/test.yml (hunk 7 above)
Comment (jrfnl) [reply .github/workflows/test.yml]: That's not really what I meant - the branch selection means that pushes to branches other than `develop` and `master` will **_not_** be checked, which means that for someone working on a branch, there is no way to check the status until they submit a pull request.

By having `push` without the branch selection, **_all_** pushes will be checked via GH Actions, not just the merges to `develop` or `master`.

On that note: this repo doesn't even have a `develop` branch, so, that branch should be removed either way.

Comment (grogy) [reply .github/workflows/test.yml]: Ahh, thank you. I got it wrong. So I will delete `branches` part

**Review by jrfnl — APPROVED** [review]

**Comment by jrfnl** [comment]
Oops, sorry, forgot to do rebase merge.....

**Pull request merged**